tests/TestCase.php

```diff
@@ -199,8 +199,52 @@ protected function postGraphQL(array $data, array $headers = []): TestResponse
     {
         return $this->postJson(
             'graphql',
-            $data,
-            $headers
+            $data
+        );
+    }
+
+    /**
+     * Send a multipart form request.
+     *
+     * This is used for file uploads conforming to the specification:
+     * https://github.com/jaydenseric/graphql-multipart-request-spec
+     *
+     * @param  mixed[]  $data
+     * @return \Illuminate\Foundation\Testing\TestResponse
+     */
+    protected function postGraphQLMultipart(array $data): TestResponse
+    {
+        // JSON encode and add parameters only if they are given.
+        // This allows tests with invalid payloads to be sent.
+        $parameters = [];
+        if (isset($data['operations'])) {
+            $parameters['operations'] = json_encode($data['operations']);
+        }
+
+        if (isset($data['map'])) {
+            $parameters['map'] = json_encode($data['map']);
+        }
+
+        // Add files
+        $files = [];
+
+        // If array is zero-index, there are files attached to the query
+        if (isset($data[0])) {
+            // Loop through all files, and add them to the payload
+            for ($i = 0; isset($data[$i]); $i++) {
```

Comment:
Why is this so complicated? Can't this be `$files = $data;`

Reply:
`$data` also contains `operations` and `map`. Would you prefer following?
```php
unset($data['operations']);
unset($data['map']);

$files = $data; // unnecessary - can just be added directly as an argument in $this->call()
```

Reply:
The keys for the files are actually arbitrary, so let's just pass them as strings.

```php
[
	'operations' => //
	'map' => //
	'0' => //
]
```

Reply:
So we are reverting to the original test?

Reply:
Just checking if this simulates the request properly in my local environment, will report back soon.